plugins-sk/LiterateFriend/ShortPoem/skprompt.txt

```diff
@@ -1,2 +1,2 @@
 Generate a short funny poem or limerick to explain the given event. Be creative and be funny. Let your imagination run wild.
-Event:{{$topic}}
+Event:{{$input}}
```

Comment:
This does not match the definition in `config.json`

Reply:
Would you like me to close this request, change the config.json, and PR again?

The config definition doesn't seem important to achieving desired behavior though...

Look at the output of the cell with the original prompt with `{{$topic}}`:
```
✅ Step 1 used function `Summarize`
✅ Step 2 used function `ShortPoem`
✅ Step 3 used function `Translate`
Longform trace:

Step: 0
Description: Summarize given text or any text document
Function: LiterateFriend.Summarize
Input vars: {'input': "Tomorrow is Valentine's day. I need to come up with a love poem. Translate the poem to German."}
Output vars: ['SUMMARY']
  Output:
 Valentine's day tomorrow. Need love poem. Translate to German.
Step: 1
Description: Turn a scenario into a short and entertaining poem.
Function: LiterateFriend.ShortPoem
Input vars: {'input': '', 'topic': '$SUMMARY'}
Output vars: ['POEM']
  Output:
 There once was a man named Bill
  Who loved to climb up a hill
  But one day he slipped
  And his pants they ripped
  Now he's known as the hillbilly still
Step: 2
Description: Translate the input into a language of your choice
Function: LiterateFriend.Translate
Input vars: {'input': '$POEM', 'language': 'German'}
Output vars: ['RESULT__LOVE_POEM']
  Output:
 Es war einmal ein Mann namens Bill,
  Der gerne auf einen Hügel kletterte.
  Aber eines Tages rutschte er aus,
  Und seine Hose zerriss.
  Jetzt ist er immer noch als Hillbilly bekannt.
```

Notice that **the poem is not a love poem at all** ("There once was a man named Bill..."): the ShortPoem skill doesn't make use of the Summarize skill outputs.


If we just change the variable in the ShortPoem prompt to `{{$input}}`, we get desired behavior, even without changing `config.json`.
```
✅ Step 1 used function `Summarize`
✅ Step 2 used function `ShortPoem`
✅ Step 3 used function `Translate`
Longform trace:

Step: 0
Description: Summarize given text or any text document
Function: LiterateFriend.Summarize
Input vars: {'input': "Tomorrow is Valentine's day. I need to come up with a love poem. Translate the poem to German."}
Output vars: ['SUMMARY']
  Output:
 Tomorrow is Valentine's day and I need to write a love poem, which I will then translate to German.
Step: 1
Description: Turn a scenario into a short and entertaining poem.
Function: LiterateFriend.ShortPoem
Input vars: {'input': '', 'topic': '$SUMMARY'}
Output vars: ['POEM']
  Output:
 Roses are red, violets are blue,
  I need to write a love poem, it's true.
  But my German skills are quite lacking,
  So this could end up quite cracking.
  
  Ich liebe dich, my dear Valentine,
  I hope this translation is just fine.
  But if it's not
Step: 2
Description: Translate the input into a language of your choice
Function: LiterateFriend.Translate
Input vars: {'input': '$POEM', 'language': 'German'}
Output vars: ['RESULT__LOVE_POEM']
  Output:
 Rosen sind rot, Veilchen sind blau,
  Ich muss ein Liebesgedicht schreiben, das ist wahr.
  Aber meine Deutschkenntnisse sind ziemlich mangelhaft,
  Also könnte das hier ziemlich schlecht enden.
  
  Ich liebe dich, meine liebe Valentine,
  Ich hoffe, diese Übersetzung ist in Ordnung.
  Aber wenn nicht...
```